### PR TITLE
feat(password) : user can now change his password

### PR DESCRIPTION
**Pull request by nathanvandeperre97**

### CODE REVIEW

File: lib/pages/change_password.dart [hunk 1]
```diff
@@ -1,11 +1,207 @@
...
+          changePassword();
+        }
+      },
+      child: Text(getString(context, 'confirm_password_changing')),
+    );
+  }
+
+  Container buildNewPassword(BuildContext context) {
+    return Container(
+      width: MediaQuery.of(context).size.width - 10,
```
Comment (zohra051) [inline lib/pages/change_password.dart]: please use : MediaQuery.of(context).size.width *

File: lib/pages/change_password.dart [hunk 2]
```diff
@@ -1,11 +1,207 @@
...
+      height: 90,
```
Comment (zohra051) [inline lib/pages/change_password.dart]: please use : MediaQuery.of(context).size.height *

File: lib/pages/change_password.dart [hunk 3]
```diff
@@ -1,11 +1,207 @@
...
+        ),
+      ),
+    );
+  }
+
+  RaisedButton buildChangePasswordButton(BuildContext context) {
+    return RaisedButton(
+      key: Key("change_password_button"),
+      color: Colors.blue,
+      textColor: Colors.white,
```
Comment (zohra051) [inline lib/pages/change_password.dart]: please use : WHITE

File: lib/pages/change_password.dart [hunk 4]
```diff
@@ -1,11 +1,207 @@
...
+          ),
+        ),
+      ),
+    );
+  }
+
+  RaisedButton buildChangePasswordButton(BuildContext context) {
+    return RaisedButton(
+      key: Key("change_password_button"),
+      color: Colors.blue,
```
Comment (zohra051) [inline lib/pages/change_password.dart]: please use : BLUE

File: lib/pages/change_password.dart [hunk 5]
```diff
@@ -1,11 +1,207 @@
...
+          if (_newPassword.length < PASSWORD_MINIMAL_LENGHT) {
+            return getString(context, 'too_short_new_password');
+          }
+          return null;
+        },
+        key: Key("new_password_label"),
+        textAlign: TextAlign.start,
+        decoration: InputDecoration(
+            enabledBorder: OutlineInputBorder(
+              borderSide: BorderSide(color: Colors.transparent),
```
Comment (zohra051) [inline lib/pages/change_password.dart]: please use : TRANSPARENT

File: lib/pages/change_password.dart [hunk 6]
```diff
@@ -1,11 +1,207 @@
...
+        controller: _newPasswordController,
+        obscureText: true,
+        onChanged: (text) => handleSubmittedNewPassword(text),
+      ),
+    );
+  }
+
+  Container buildConfirmNewPassword(BuildContext context) {
+    return Container(
+      width: MediaQuery.of(context).size.width - 10,
```
Comment (zohra051) [inline lib/pages/change_password.dart]: please use : MediaQuery.of(context).size.width *

File: lib/pages/change_password.dart [hunk 7]
```diff
@@ -1,11 +1,207 @@
...
+      height: 90,
```
Comment (zohra051) [inline lib/pages/change_password.dart]: please use : MediaQuery.of(context).size.height *

File: lib/pages/change_password.dart [hunk 8]
```diff
@@ -1,11 +1,207 @@
...
+          if (_newPassword != _confirmNewPassword) {
+            return getString(context, 'new_password_not_confirmed');
+          }
+          return null;
+        },
+        key: Key("confirm_new_password_label"),
+        textAlign: TextAlign.start,
+        decoration: InputDecoration(
+            enabledBorder: OutlineInputBorder(
+              borderSide: BorderSide(color: Colors.transparent),
```
Comment (zohra051) [inline lib/pages/change_password.dart]: please use : TRANSPARENT

File: lib/pages/change_password.dart [hunk 9]
```diff
@@ -1,11 +1,207 @@
...
+        obscureText: true,
+        onChanged: (text) => handleSubmittedConfirmNewPassword(text),
+      ),
+    );
+  }
+
+  Future<void> changePassword() async {
+    FocusScope.of(context).unfocus();
+    String result = await GetIt.I
+        .get<AuthenticationService>()
```
Comment (zohra051) [inline lib/pages/change_password.dart]: please use :   var authService = GetIt.I.get<AuthenticationService>();

File: lib/pages/change_password.dart [hunk 10]
```diff
@@ -1,11 +1,207 @@
...
+        .changePassword(GetIt.I.get<UserService>().user, _newPassword);
```
Comment (zohra051) [inline lib/pages/change_password.dart]: please use :   var userService = GetIt.I.get<UserService>();

**Review by zohra051 — COMMENTED** [review]
please change

**Review by PiotrFLEURY — APPROVED** [review]

**Pull request merged**